lib/tapioca/gemfile.rb

```diff
@@ -125,11 +131,25 @@ def rbi_file_name
 
       sig { params(path: String).returns(T::Boolean) }
       def contains_path?(path)
-        to_realpath(path).start_with?(full_gem_path) || has_parent_gemspec?(path)
+        if default_gem?
+          files.any? { |file| file.to_s == to_realpath(path) }
+        else
+          to_realpath(path).start_with?(full_gem_path) || has_parent_gemspec?(path)
+        end
       end
 
       private
 
+      sig { returns(T::Boolean) }
+      def default_gem?
+        @spec.respond_to?(:default_gem?) && @spec.default_gem?
```

Comment:
Is the `respond_to?` for older versions of Bundler? When would it not respond to `default_gem?`?

Reply:
I think so. To be perfectly honest, I cargo-culted it from [some of the Bundler code](https://github.com/rubygems/rubygems/blob/96e5cff3df491c4d943186804c6d03b57fcb459b/bundler/lib/bundler/rubygems_integration.rb#L94) that was doing similar checks so I assumed it could be an older version Rubygems, maybe.